### PR TITLE
Create endpoint to revoke token for ticket #197.

### PR DESCRIPTION
**Pull request by Buwujiu**

### CODE REVIEW

**Comment by Buwujiu** [comment]
ticket #197

File: src/main/java/bio/overture/ego/service/UserService.java [hunk 1]
```diff
@@ -530,4 +530,12 @@ protected void correctUserData(@MappingTarget User userToUpdate) {
       }
     }
   }
+
+  public boolean isActiveUser(User user) {
+    return "approved".equals(user.getStatus().toLowerCase());
+  }
+
+  public boolean isAdmin(User user) {
+    return "admin".equals((user.getRole().toLowerCase()));
```
Comment (andricDu) [inline src/main/java/bio/overture/ego/service/UserService.java]:
```
import static bio.overture.ego.model.enums.UserRole.ADMIN;
import static bio.overture.ego.model.enums.UserRole.resolveUserRoleIgnoreCase;

...

  public boolean isAdmin(User user) {
    return resolveUserRoleIgnoreCase(user.getRole()) == ADMIN;
  }

```

This way we also get type safety.

File: src/main/java/bio/overture/ego/service/TokenService.java [hunk 2]
```diff
@@ -52,33 +53,56 @@
...
+  /*
+   * Dependencies
+   */
+  private TokenSigner tokenSigner;
+  private UserService userService;
+  private ApplicationService applicationService;
+  private UserEvents userEvents;
+  private TokenStoreService tokenStoreService;
+  private PolicyService policyService;
+  private TokenStoreRepository tokenStoreRepository;
```
[the remark this answers is not in the document]
Comment (Buwujiu) [reply src/main/java/bio/overture/ego/service/TokenService.java]: tokenStoreRepository is used to construct super class at line 98:  super(Token.class, tokenStoreRepository);

[the remark this answers is not in the document]
Comment (andricDu) [reply src/main/java/bio/overture/ego/service/TokenService.java]: Right, but its never maintained as part of the state. 

You don't need this line:
```java
this.tokenStoreRepository = tokenStoreRepository;
```

You only need:
```java
super(Token.class, tokenStoreRepository);
```

[the remark this answers is not in the document]
Comment (Buwujiu) [reply src/main/java/bio/overture/ego/service/TokenService.java]: Yes you are right!

**Comment by andricDu** [comment]
Thanks @Buwujiu !

**Pull request merged**